test/e2e/azure_securitygroups.go

```diff
@@ -168,7 +169,7 @@ func AzureSecurityGroupsSpec(ctx context.Context, inputGetter func() AzureSecuri
 	By("Deleting security rule from the subnet")
 	Expect(len(expectedSubnets)).To(Not(Equal(0)))
 	testSubnet.SecurityGroup.SecurityRules = infrav1.SecurityRules{testSecurityRule2}
-	expectedSubnets = originalSubnets
+	expectedSubnets = infrav1.Subnets{}
```

Comment:
Why is this change needed? I think this would end up deleting the original subnets from the `NetworkSpec`

Reply:
Agree, I think we should leave all the other subnets untouched for this test.

I suppose an alternative to reconstructing the entire list of subnets every time is to add the one we'll modify for the test, note its index in the list, then modify it in place.

Reply:
yeah my bad I confused how the append was working (subnets vs rules)

Reply:
After going through the flow again I now understand how this was implemented (and how my tweaks were breaking things).

I'm not entirely sure why the new `ConsistsOf` usage is more correct than the prior `ContainElements`, but I am convinced from reading the docs that `ConsistsOf` should be a valid solution.

Reply:
> I'm not entirely sure why the new `ConsistsOf` usage is more correct than the prior `ContainElements`, but I am convinced from reading the docs that `ConsistsOf` should be a valid solution.

The previous check allows the "expected" list to contain a superset of the "actual" names, whereas `ConsistsOf` would check that they match exactly. @willie-yao Is there a reason we aren't doing a more strict comparison currently?

Reply:
I'm convinced, let's validate that this change passes AKS E2E and then I think we should merge it